src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -299,28 +309,28 @@ private BoundExpression VisitCollectionBuilderCollectionExpression(BoundCollecti
             return result;
         }
 
-        internal static bool ShouldUseRuntimeHelpersCreateSpan(BoundCollectionExpression node, TypeSymbol elementType)
+        internal static bool IsAllocatingRefStructCollectionExpression(BoundCollectionExpressionBase node, TypeSymbol targetType, TypeSymbol? elementType, CSharpCompilation compilation)
         {
-            Debug.Assert(node.CollectionTypeKind is
-                CollectionExpressionTypeKind.ReadOnlySpan or
-                CollectionExpressionTypeKind.CollectionBuilder);
+            return targetType.IsRefLikeType
```

Comment:
>targetType.IsRefLikeType [](http://example.com/codeflow?start=19&length=24)

Should this be spans only? The warning seems confusing for other `ref struct` types since it's not clear how to use an array in those cases.

Reply:
Maybe. It would definitely simplify impl here while satisfying the vast majority of usage patterns that we expect.